### PR TITLE
pdb: default to maxUnavailable=1 instead of minAvailable=1

### PR DESCRIPTION
**Pull request by consideRatio**
If you want to lower the replicas of user-scheduler to 1, then you will get the same PDB issues we have had with hub/proxy (#1938) where they block upgrades etc because we have `minAvailable: 1` instead of `maxUnavailable: 1` in order to handle workloads.

### CODE REVIEW

File: jupyterhub/schema.yaml [hunk 1]
```diff
@@ -890,9 +890,12 @@ properties:
             description: |
               Decides if a PodDisruptionBudget is created targeting the
               Deployment's pods.
+          maxUnavailable:
+            type: [integer, "null"]
+            description: |
+              The maximum number of pods required to not be available during
```
Comment (yuvipanda) [inline jupyterhub/schema.yaml:896]:
```suggestion
              The maximum number of pods that can be unavailable during
```

**Comment by yuvipanda** [comment]
So if we have 2 replicas, it will still ensure there will at least be one available, right?

**Comment by consideRatio** [comment]
> So if we have 2 replicas, it will still ensure there will at least be one available, right?

Yepp! Note btw that maxUnavailable was introduced in k8s 1.9 or something, which was later than the PDBs were introduced in the first place in z2jh.

**Comment by consideRatio** [comment]
image

Hmmmmm, perhaps did a mistake with the logic... It should show maxUnavailable there.

**Comment by consideRatio** [comment]
Omg, `kindOf` returns `"float64"` instead of `"int"`, woops.

**Comment by consideRatio** [comment]
Okay finally got everything right, the tests does a great job now! :heart:

**Comment by yuvipanda** [comment]
Thanks, @consideRatio!

**Comment by consideRatio** [comment]
Thanks @yuvipanda for your help with review/merge!

**Pull request merged**